[2024-12-08T21:40:13.590+0000] {local_task_job_runner.py:123} INFO - ::group::Pre task execution logs
[2024-12-08T21:40:13.605+0000] {taskinstance.py:2613} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: sparking_flow.python_job manual__2024-12-08T21:40:11.545590+00:00 [queued]>
[2024-12-08T21:40:13.612+0000] {taskinstance.py:2613} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: sparking_flow.python_job manual__2024-12-08T21:40:11.545590+00:00 [queued]>
[2024-12-08T21:40:13.613+0000] {taskinstance.py:2866} INFO - Starting attempt 1 of 1
[2024-12-08T21:40:13.623+0000] {taskinstance.py:2889} INFO - Executing <Task(SparkSubmitOperator): python_job> on 2024-12-08 21:40:11.545590+00:00
[2024-12-08T21:40:13.630+0000] {logging_mixin.py:190} WARNING - /home/***/.local/lib/python3.12/site-packages/***/task/task_runner/standard_task_runner.py:70 DeprecationWarning: This process (pid=184) is multi-threaded, use of fork() may lead to deadlocks in the child.
[2024-12-08T21:40:13.631+0000] {standard_task_runner.py:72} INFO - Started process 189 to run task
[2024-12-08T21:40:13.630+0000] {standard_task_runner.py:104} INFO - Running: ['***', 'tasks', 'run', 'sparking_flow', 'python_job', 'manual__2024-12-08T21:40:11.545590+00:00', '--job-id', '271', '--raw', '--subdir', 'DAGS_FOLDER/spark_***.py', '--cfg-path', '/tmp/tmp506tbd3w']
[2024-12-08T21:40:13.633+0000] {standard_task_runner.py:105} INFO - Job 271: Subtask python_job
[2024-12-08T21:40:13.681+0000] {task_command.py:467} INFO - Running <TaskInstance: sparking_flow.python_job manual__2024-12-08T21:40:11.545590+00:00 [running]> on host 2a892a3469ec
[2024-12-08T21:40:13.762+0000] {taskinstance.py:3132} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='Yusuf Ganiyu' AIRFLOW_CTX_DAG_ID='sparking_flow' AIRFLOW_CTX_TASK_ID='python_job' AIRFLOW_CTX_EXECUTION_DATE='2024-12-08T21:40:11.545590+00:00' AIRFLOW_CTX_TRY_NUMBER='1' AIRFLOW_CTX_DAG_RUN_ID='manual__2024-12-08T21:40:11.545590+00:00'
[2024-12-08T21:40:13.762+0000] {taskinstance.py:731} INFO - ::endgroup::
[2024-12-08T21:40:13.784+0000] {base.py:84} INFO - Retrieving connection 'spark-conn'
[2024-12-08T21:40:13.786+0000] {spark_submit.py:473} INFO - Spark-Submit cmd: spark-submit --master spark://sparkingflow-spark-master-1:7077 --name arrow-spark --deploy-mode client jobs/python/wordcountjob.py
[2024-12-08T21:40:19.158+0000] {spark_submit.py:634} INFO - 24/12/08 21:40:19 INFO SparkContext: Running Spark version 3.5.3
[2024-12-08T21:40:19.167+0000] {spark_submit.py:634} INFO - 24/12/08 21:40:19 INFO SparkContext: OS info Linux, 6.10.14-linuxkit, aarch64
[2024-12-08T21:40:19.171+0000] {spark_submit.py:634} INFO - 24/12/08 21:40:19 INFO SparkContext: Java version 17.0.13
[2024-12-08T21:40:19.271+0000] {spark_submit.py:634} INFO - 24/12/08 21:40:19 WARN NativeCodeLoader: Unable to load native-hadoop library for your platform... using builtin-java classes where applicable
[2024-12-08T21:40:19.492+0000] {spark_submit.py:634} INFO - 24/12/08 21:40:19 INFO ResourceUtils: ==============================================================
[2024-12-08T21:40:19.498+0000] {spark_submit.py:634} INFO - 24/12/08 21:40:19 INFO ResourceUtils: No custom resources configured for spark.driver.
[2024-12-08T21:40:19.499+0000] {spark_submit.py:634} INFO - 24/12/08 21:40:19 INFO ResourceUtils: ==============================================================
[2024-12-08T21:40:19.501+0000] {spark_submit.py:634} INFO - 24/12/08 21:40:19 INFO SparkContext: Submitted application: CSVReadExample
[2024-12-08T21:40:19.539+0000] {spark_submit.py:634} INFO - 24/12/08 21:40:19 INFO ResourceProfile: Default ResourceProfile created, executor resources: Map(memory -> name: memory, amount: 1024, script: , vendor: , offHeap -> name: offHeap, amount: 0, script: , vendor: ), task resources: Map(cpus -> name: cpus, amount: 1.0)
[2024-12-08T21:40:19.559+0000] {spark_submit.py:634} INFO - 24/12/08 21:40:19 INFO ResourceProfile: Limiting resource is cpu
[2024-12-08T21:40:19.563+0000] {spark_submit.py:634} INFO - 24/12/08 21:40:19 INFO ResourceProfileManager: Added ResourceProfile id: 0
[2024-12-08T21:40:19.645+0000] {spark_submit.py:634} INFO - 24/12/08 21:40:19 INFO SecurityManager: Changing view acls to: ***
[2024-12-08T21:40:19.647+0000] {spark_submit.py:634} INFO - 24/12/08 21:40:19 INFO SecurityManager: Changing modify acls to: ***
[2024-12-08T21:40:19.648+0000] {spark_submit.py:634} INFO - 24/12/08 21:40:19 INFO SecurityManager: Changing view acls groups to:
[2024-12-08T21:40:19.655+0000] {spark_submit.py:634} INFO - 24/12/08 21:40:19 INFO SecurityManager: Changing modify acls groups to:
[2024-12-08T21:40:19.656+0000] {spark_submit.py:634} INFO - 24/12/08 21:40:19 INFO SecurityManager: SecurityManager: authentication disabled; ui acls disabled; users with view permissions: ***; groups with view permissions: EMPTY; users with modify permissions: ***; groups with modify permissions: EMPTY
[2024-12-08T21:40:20.242+0000] {spark_submit.py:634} INFO - 24/12/08 21:40:20 INFO Utils: Successfully started service 'sparkDriver' on port 45397.
[2024-12-08T21:40:20.301+0000] {spark_submit.py:634} INFO - 24/12/08 21:40:20 INFO SparkEnv: Registering MapOutputTracker
[2024-12-08T21:40:20.355+0000] {spark_submit.py:634} INFO - 24/12/08 21:40:20 INFO SparkEnv: Registering BlockManagerMaster
[2024-12-08T21:40:20.382+0000] {spark_submit.py:634} INFO - 24/12/08 21:40:20 INFO BlockManagerMasterEndpoint: Using org.apache.spark.storage.DefaultTopologyMapper for getting topology information
[2024-12-08T21:40:20.387+0000] {spark_submit.py:634} INFO - 24/12/08 21:40:20 INFO BlockManagerMasterEndpoint: BlockManagerMasterEndpoint up
[2024-12-08T21:40:20.397+0000] {spark_submit.py:634} INFO - 24/12/08 21:40:20 INFO SparkEnv: Registering BlockManagerMasterHeartbeat
[2024-12-08T21:40:20.446+0000] {spark_submit.py:634} INFO - 24/12/08 21:40:20 INFO DiskBlockManager: Created local directory at /tmp/blockmgr-5642b523-e0cd-4de8-a7f4-72cd48960ac8
[2024-12-08T21:40:20.483+0000] {spark_submit.py:634} INFO - 24/12/08 21:40:20 INFO MemoryStore: MemoryStore started with capacity 434.4 MiB
[2024-12-08T21:40:20.530+0000] {spark_submit.py:634} INFO - 24/12/08 21:40:20 INFO SparkEnv: Registering OutputCommitCoordinator
[2024-12-08T21:40:20.832+0000] {spark_submit.py:634} INFO - 24/12/08 21:40:20 INFO JettyUtils: Start Jetty 0.0.0.0:4040 for SparkUI
[2024-12-08T21:40:20.964+0000] {spark_submit.py:634} INFO - 24/12/08 21:40:20 WARN Utils: Service 'SparkUI' could not bind on port 4040. Attempting port 4041.
[2024-12-08T21:40:20.966+0000] {spark_submit.py:634} INFO - 24/12/08 21:40:20 WARN Utils: Service 'SparkUI' could not bind on port 4041. Attempting port 4042.
[2024-12-08T21:40:20.985+0000] {spark_submit.py:634} INFO - 24/12/08 21:40:20 INFO Utils: Successfully started service 'SparkUI' on port 4042.
[2024-12-08T21:40:21.224+0000] {spark_submit.py:634} INFO - 24/12/08 21:40:21 INFO StandaloneAppClient$ClientEndpoint: Connecting to master spark://sparkingflow-spark-master-1:7077...
[2024-12-08T21:40:21.363+0000] {spark_submit.py:634} INFO - 24/12/08 21:40:21 INFO TransportClientFactory: Successfully created connection to sparkingflow-spark-master-1/172.18.0.5:7077 after 64 ms (0 ms spent in bootstraps)
[2024-12-08T21:40:21.670+0000] {spark_submit.py:634} INFO - 24/12/08 21:40:21 INFO StandaloneSchedulerBackend: Connected to Spark cluster with app ID app-20241208214021-0001
[2024-12-08T21:40:21.694+0000] {spark_submit.py:634} INFO - 24/12/08 21:40:21 INFO Utils: Successfully started service 'org.apache.spark.network.netty.NettyBlockTransferService' on port 38959.
[2024-12-08T21:40:21.698+0000] {spark_submit.py:634} INFO - 24/12/08 21:40:21 INFO NettyBlockTransferService: Server created on 2a892a3469ec:38959
[2024-12-08T21:40:21.700+0000] {spark_submit.py:634} INFO - 24/12/08 21:40:21 INFO BlockManager: Using org.apache.spark.storage.RandomBlockReplicationPolicy for block replication policy
[2024-12-08T21:40:21.710+0000] {spark_submit.py:634} INFO - 24/12/08 21:40:21 INFO BlockManagerMaster: Registering BlockManager BlockManagerId(driver, 2a892a3469ec, 38959, None)
[2024-12-08T21:40:21.717+0000] {spark_submit.py:634} INFO - 24/12/08 21:40:21 INFO BlockManagerMasterEndpoint: Registering block manager 2a892a3469ec:38959 with 434.4 MiB RAM, BlockManagerId(driver, 2a892a3469ec, 38959, None)
[2024-12-08T21:40:21.719+0000] {spark_submit.py:634} INFO - 24/12/08 21:40:21 INFO BlockManagerMaster: Registered BlockManager BlockManagerId(driver, 2a892a3469ec, 38959, None)
[2024-12-08T21:40:21.721+0000] {spark_submit.py:634} INFO - 24/12/08 21:40:21 INFO BlockManager: Initialized BlockManager: BlockManagerId(driver, 2a892a3469ec, 38959, None)
[2024-12-08T21:40:22.165+0000] {spark_submit.py:634} INFO - 24/12/08 21:40:22 INFO StandaloneSchedulerBackend: SchedulerBackend is ready for scheduling beginning after reached minRegisteredResourcesRatio: 0.0
[2024-12-08T21:40:22.565+0000] {spark_submit.py:634} INFO - 24/12/08 21:40:22 INFO SparkContext: Added file /tmp/tmpfucegs0h/test1234.csv at spark://2a892a3469ec:45397/files/test1234.csv with timestamp 1733694022563
[2024-12-08T21:40:22.566+0000] {spark_submit.py:634} INFO - 24/12/08 21:40:22 INFO Utils: Copying /tmp/tmpfucegs0h/test1234.csv to /tmp/spark-d647b6e1-10af-46db-9915-183687f7eab0/userFiles-b0c4d8f8-2b5b-44b3-8f66-ae1674e6840f/test1234.csv
[2024-12-08T21:40:22.614+0000] {spark_submit.py:634} INFO - 24/12/08 21:40:22 INFO SharedState: Setting hive.metastore.warehouse.dir ('null') to the value of spark.sql.warehouse.dir.
[2024-12-08T21:40:22.617+0000] {spark_submit.py:634} INFO - 24/12/08 21:40:22 INFO SharedState: Warehouse path is 'file:/opt/***/spark-warehouse'.
[2024-12-08T21:40:23.585+0000] {spark_submit.py:634} INFO - 24/12/08 21:40:23 INFO InMemoryFileIndex: It took 72 ms to list leaf files for 1 paths.
[2024-12-08T21:40:23.672+0000] {spark_submit.py:634} INFO - 24/12/08 21:40:23 INFO InMemoryFileIndex: It took 2 ms to list leaf files for 1 paths.
[2024-12-08T21:40:25.582+0000] {spark_submit.py:634} INFO - 24/12/08 21:40:25 INFO StandaloneAppClient$ClientEndpoint: Executor added: app-20241208214021-0001/0 on worker-20241208213948-172.18.0.7-44039 (172.18.0.7:44039) with 2 core(s)
[2024-12-08T21:40:25.601+0000] {spark_submit.py:634} INFO - 24/12/08 21:40:25 INFO StandaloneSchedulerBackend: Granted executor ID app-20241208214021-0001/0 on hostPort 172.18.0.7:44039 with 2 core(s), 1024.0 MiB RAM
[2024-12-08T21:40:25.695+0000] {spark_submit.py:634} INFO - 24/12/08 21:40:25 INFO StandaloneAppClient$ClientEndpoint: Executor updated: app-20241208214021-0001/0 is now RUNNING
[2024-12-08T21:40:26.852+0000] {spark_submit.py:634} INFO - 24/12/08 21:40:26 INFO FileSourceStrategy: Pushed Filters:
[2024-12-08T21:40:26.858+0000] {spark_submit.py:634} INFO - 24/12/08 21:40:26 INFO FileSourceStrategy: Post-Scan Filters: (length(trim(value#0, None)) > 0)
[2024-12-08T21:40:27.579+0000] {spark_submit.py:634} INFO - 24/12/08 21:40:27 INFO CodeGenerator: Code generated in 242.139625 ms
[2024-12-08T21:40:27.634+0000] {spark_submit.py:634} INFO - 24/12/08 21:40:27 INFO MemoryStore: Block broadcast_0 stored as values in memory (estimated size 199.6 KiB, free 434.2 MiB)
[2024-12-08T21:40:27.709+0000] {spark_submit.py:634} INFO - 24/12/08 21:40:27 INFO MemoryStore: Block broadcast_0_piece0 stored as bytes in memory (estimated size 34.4 KiB, free 434.2 MiB)
[2024-12-08T21:40:27.721+0000] {spark_submit.py:634} INFO - 24/12/08 21:40:27 INFO BlockManagerInfo: Added broadcast_0_piece0 in memory on 2a892a3469ec:38959 (size: 34.4 KiB, free: 434.4 MiB)
[2024-12-08T21:40:27.730+0000] {spark_submit.py:634} INFO - 24/12/08 21:40:27 INFO SparkContext: Created broadcast 0 from csv at NativeMethodAccessorImpl.java:0
[2024-12-08T21:40:27.752+0000] {spark_submit.py:634} INFO - 24/12/08 21:40:27 INFO FileSourceScanExec: Planning scan with bin packing, max size: 4194304 bytes, open cost is considered as scanning 4194304 bytes.
[2024-12-08T21:40:27.864+0000] {spark_submit.py:634} INFO - 24/12/08 21:40:27 INFO SparkContext: Starting job: csv at NativeMethodAccessorImpl.java:0
[2024-12-08T21:40:27.888+0000] {spark_submit.py:634} INFO - 24/12/08 21:40:27 INFO DAGScheduler: Got job 0 (csv at NativeMethodAccessorImpl.java:0) with 1 output partitions
[2024-12-08T21:40:27.888+0000] {spark_submit.py:634} INFO - 24/12/08 21:40:27 INFO DAGScheduler: Final stage: ResultStage 0 (csv at NativeMethodAccessorImpl.java:0)
[2024-12-08T21:40:27.889+0000] {spark_submit.py:634} INFO - 24/12/08 21:40:27 INFO DAGScheduler: Parents of final stage: List()
[2024-12-08T21:40:27.889+0000] {spark_submit.py:634} INFO - 24/12/08 21:40:27 INFO DAGScheduler: Missing parents: List()
[2024-12-08T21:40:27.892+0000] {spark_submit.py:634} INFO - 24/12/08 21:40:27 INFO DAGScheduler: Submitting ResultStage 0 (MapPartitionsRDD[3] at csv at NativeMethodAccessorImpl.java:0), which has no missing parents
[2024-12-08T21:40:27.962+0000] {spark_submit.py:634} INFO - 24/12/08 21:40:27 INFO MemoryStore: Block broadcast_1 stored as values in memory (estimated size 13.5 KiB, free 434.2 MiB)
[2024-12-08T21:40:27.970+0000] {spark_submit.py:634} INFO - 24/12/08 21:40:27 INFO MemoryStore: Block broadcast_1_piece0 stored as bytes in memory (estimated size 6.4 KiB, free 434.2 MiB)
[2024-12-08T21:40:27.971+0000] {spark_submit.py:634} INFO - 24/12/08 21:40:27 INFO BlockManagerInfo: Added broadcast_1_piece0 in memory on 2a892a3469ec:38959 (size: 6.4 KiB, free: 434.4 MiB)
[2024-12-08T21:40:27.976+0000] {spark_submit.py:634} INFO - 24/12/08 21:40:27 INFO SparkContext: Created broadcast 1 from broadcast at DAGScheduler.scala:1585
[2024-12-08T21:40:28.021+0000] {spark_submit.py:634} INFO - 24/12/08 21:40:28 INFO DAGScheduler: Submitting 1 missing tasks from ResultStage 0 (MapPartitionsRDD[3] at csv at NativeMethodAccessorImpl.java:0) (first 15 tasks are for partitions Vector(0))
[2024-12-08T21:40:28.023+0000] {spark_submit.py:634} INFO - 24/12/08 21:40:28 INFO TaskSchedulerImpl: Adding task set 0.0 with 1 tasks resource profile 0
[2024-12-08T21:40:28.989+0000] {spark_submit.py:634} INFO - 24/12/08 21:40:28 INFO StandaloneSchedulerBackend$StandaloneDriverEndpoint: Registered executor NettyRpcEndpointRef(spark-client://Executor) (172.18.0.7:33206) with ID 0,  ResourceProfileId 0
[2024-12-08T21:40:29.038+0000] {spark_submit.py:634} INFO - 24/12/08 21:40:29 INFO BlockManagerMasterEndpoint: Registering block manager 172.18.0.7:35165 with 434.4 MiB RAM, BlockManagerId(0, 172.18.0.7, 35165, None)
[2024-12-08T21:40:29.088+0000] {spark_submit.py:634} INFO - 24/12/08 21:40:29 INFO TaskSetManager: Starting task 0.0 in stage 0.0 (TID 0) (172.18.0.7, executor 0, partition 0, PROCESS_LOCAL, 9852 bytes)
[2024-12-08T21:40:29.301+0000] {spark_submit.py:634} INFO - 24/12/08 21:40:29 INFO BlockManagerInfo: Added broadcast_1_piece0 in memory on 172.18.0.7:35165 (size: 6.4 KiB, free: 434.4 MiB)
[2024-12-08T21:40:29.990+0000] {spark_submit.py:634} INFO - 24/12/08 21:40:29 INFO BlockManagerInfo: Added broadcast_0_piece0 in memory on 172.18.0.7:35165 (size: 34.4 KiB, free: 434.4 MiB)
[2024-12-08T21:40:30.342+0000] {spark_submit.py:634} INFO - 24/12/08 21:40:30 WARN TaskSetManager: Lost task 0.0 in stage 0.0 (TID 0) (172.18.0.7 executor 0): org.apache.spark.SparkFileNotFoundException: File file:/tmp/spark-d647b6e1-10af-46db-9915-183687f7eab0/userFiles-b0c4d8f8-2b5b-44b3-8f66-ae1674e6840f/test1234.csv does not exist
[2024-12-08T21:40:30.342+0000] {spark_submit.py:634} INFO - It is possible the underlying files have been updated. You can explicitly invalidate the cache in Spark by running 'REFRESH TABLE tableName' command in SQL or by recreating the Dataset/DataFrame involved.
[2024-12-08T21:40:30.343+0000] {spark_submit.py:634} INFO - at org.apache.spark.sql.errors.QueryExecutionErrors$.readCurrentFileNotFoundError(QueryExecutionErrors.scala:781)
[2024-12-08T21:40:30.343+0000] {spark_submit.py:634} INFO - at org.apache.spark.sql.execution.datasources.FileScanRDD$$anon$1.org$apache$spark$sql$execution$datasources$FileScanRDD$$anon$$readCurrentFile(FileScanRDD.scala:220)
[2024-12-08T21:40:30.343+0000] {spark_submit.py:634} INFO - at org.apache.spark.sql.execution.datasources.FileScanRDD$$anon$1.nextIterator(FileScanRDD.scala:279)
[2024-12-08T21:40:30.343+0000] {spark_submit.py:634} INFO - at org.apache.spark.sql.execution.datasources.FileScanRDD$$anon$1.hasNext(FileScanRDD.scala:129)
[2024-12-08T21:40:30.344+0000] {spark_submit.py:634} INFO - at scala.collection.Iterator$$anon$10.hasNext(Iterator.scala:460)
[2024-12-08T21:40:30.344+0000] {spark_submit.py:634} INFO - at org.apache.spark.sql.catalyst.expressions.GeneratedClass$GeneratedIteratorForCodegenStage1.processNext(Unknown Source)
[2024-12-08T21:40:30.344+0000] {spark_submit.py:634} INFO - at org.apache.spark.sql.execution.BufferedRowIterator.hasNext(BufferedRowIterator.java:43)
[2024-12-08T21:40:30.344+0000] {spark_submit.py:634} INFO - at org.apache.spark.sql.execution.WholeStageCodegenEvaluatorFactory$WholeStageCodegenPartitionEvaluator$$anon$1.hasNext(WholeStageCodegenEvaluatorFactory.scala:43)
[2024-12-08T21:40:30.345+0000] {spark_submit.py:634} INFO - at org.apache.spark.sql.execution.SparkPlan.$anonfun$getByteArrayRdd$1(SparkPlan.scala:388)
[2024-12-08T21:40:30.345+0000] {spark_submit.py:634} INFO - at org.apache.spark.rdd.RDD.$anonfun$mapPartitionsInternal$2(RDD.scala:893)
[2024-12-08T21:40:30.345+0000] {spark_submit.py:634} INFO - at org.apache.spark.rdd.RDD.$anonfun$mapPartitionsInternal$2$adapted(RDD.scala:893)
[2024-12-08T21:40:30.345+0000] {spark_submit.py:634} INFO - at org.apache.spark.rdd.MapPartitionsRDD.compute(MapPartitionsRDD.scala:52)
[2024-12-08T21:40:30.345+0000] {spark_submit.py:634} INFO - at org.apache.spark.rdd.RDD.computeOrReadCheckpoint(RDD.scala:367)
[2024-12-08T21:40:30.346+0000] {spark_submit.py:634} INFO - at org.apache.spark.rdd.RDD.iterator(RDD.scala:331)
[2024-12-08T21:40:30.346+0000] {spark_submit.py:634} INFO - at org.apache.spark.scheduler.ResultTask.runTask(ResultTask.scala:93)
[2024-12-08T21:40:30.346+0000] {spark_submit.py:634} INFO - at org.apache.spark.TaskContext.runTaskWithListeners(TaskContext.scala:166)
[2024-12-08T21:40:30.346+0000] {spark_submit.py:634} INFO - at org.apache.spark.scheduler.Task.run(Task.scala:141)
[2024-12-08T21:40:30.347+0000] {spark_submit.py:634} INFO - at org.apache.spark.executor.Executor$TaskRunner.$anonfun$run$4(Executor.scala:620)
[2024-12-08T21:40:30.347+0000] {spark_submit.py:634} INFO - at org.apache.spark.util.SparkErrorUtils.tryWithSafeFinally(SparkErrorUtils.scala:64)
[2024-12-08T21:40:30.347+0000] {spark_submit.py:634} INFO - at org.apache.spark.util.SparkErrorUtils.tryWithSafeFinally$(SparkErrorUtils.scala:61)
[2024-12-08T21:40:30.347+0000] {spark_submit.py:634} INFO - at org.apache.spark.util.Utils$.tryWithSafeFinally(Utils.scala:94)
[2024-12-08T21:40:30.348+0000] {spark_submit.py:634} INFO - at org.apache.spark.executor.Executor$TaskRunner.run(Executor.scala:623)
[2024-12-08T21:40:30.348+0000] {spark_submit.py:634} INFO - at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
[2024-12-08T21:40:30.349+0000] {spark_submit.py:634} INFO - at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
[2024-12-08T21:40:30.349+0000] {spark_submit.py:634} INFO - at java.base/java.lang.Thread.run(Thread.java:840)
[2024-12-08T21:40:30.349+0000] {spark_submit.py:634} INFO - 
[2024-12-08T21:40:30.350+0000] {spark_submit.py:634} INFO - 24/12/08 21:40:30 INFO TaskSetManager: Starting task 0.1 in stage 0.0 (TID 1) (172.18.0.7, executor 0, partition 0, PROCESS_LOCAL, 9852 bytes)
[2024-12-08T21:40:30.377+0000] {spark_submit.py:634} INFO - 24/12/08 21:40:30 INFO TaskSetManager: Lost task 0.1 in stage 0.0 (TID 1) on 172.18.0.7, executor 0: org.apache.spark.SparkFileNotFoundException (File file:/tmp/spark-d647b6e1-10af-46db-9915-183687f7eab0/userFiles-b0c4d8f8-2b5b-44b3-8f66-ae1674e6840f/test1234.csv does not exist
[2024-12-08T21:40:30.377+0000] {spark_submit.py:634} INFO - It is possible the underlying files have been updated. You can explicitly invalidate the cache in Spark by running 'REFRESH TABLE tableName' command in SQL or by recreating the Dataset/DataFrame involved.) [duplicate 1]
[2024-12-08T21:40:30.378+0000] {spark_submit.py:634} INFO - 24/12/08 21:40:30 INFO TaskSetManager: Starting task 0.2 in stage 0.0 (TID 2) (172.18.0.7, executor 0, partition 0, PROCESS_LOCAL, 9852 bytes)
[2024-12-08T21:40:30.398+0000] {spark_submit.py:634} INFO - 24/12/08 21:40:30 INFO TaskSetManager: Lost task 0.2 in stage 0.0 (TID 2) on 172.18.0.7, executor 0: org.apache.spark.SparkFileNotFoundException (File file:/tmp/spark-d647b6e1-10af-46db-9915-183687f7eab0/userFiles-b0c4d8f8-2b5b-44b3-8f66-ae1674e6840f/test1234.csv does not exist
[2024-12-08T21:40:30.398+0000] {spark_submit.py:634} INFO - It is possible the underlying files have been updated. You can explicitly invalidate the cache in Spark by running 'REFRESH TABLE tableName' command in SQL or by recreating the Dataset/DataFrame involved.) [duplicate 2]
[2024-12-08T21:40:30.399+0000] {spark_submit.py:634} INFO - 24/12/08 21:40:30 INFO TaskSetManager: Starting task 0.3 in stage 0.0 (TID 3) (172.18.0.7, executor 0, partition 0, PROCESS_LOCAL, 9852 bytes)
[2024-12-08T21:40:30.419+0000] {spark_submit.py:634} INFO - 24/12/08 21:40:30 INFO TaskSetManager: Lost task 0.3 in stage 0.0 (TID 3) on 172.18.0.7, executor 0: org.apache.spark.SparkFileNotFoundException (File file:/tmp/spark-d647b6e1-10af-46db-9915-183687f7eab0/userFiles-b0c4d8f8-2b5b-44b3-8f66-ae1674e6840f/test1234.csv does not exist
[2024-12-08T21:40:30.420+0000] {spark_submit.py:634} INFO - It is possible the underlying files have been updated. You can explicitly invalidate the cache in Spark by running 'REFRESH TABLE tableName' command in SQL or by recreating the Dataset/DataFrame involved.) [duplicate 3]
[2024-12-08T21:40:30.421+0000] {spark_submit.py:634} INFO - 24/12/08 21:40:30 ERROR TaskSetManager: Task 0 in stage 0.0 failed 4 times; aborting job
[2024-12-08T21:40:30.421+0000] {spark_submit.py:634} INFO - 24/12/08 21:40:30 INFO TaskSchedulerImpl: Removed TaskSet 0.0, whose tasks have all completed, from pool
[2024-12-08T21:40:30.423+0000] {spark_submit.py:634} INFO - 24/12/08 21:40:30 INFO TaskSchedulerImpl: Cancelling stage 0
[2024-12-08T21:40:30.424+0000] {spark_submit.py:634} INFO - 24/12/08 21:40:30 INFO TaskSchedulerImpl: Killing all running tasks in stage 0: Stage cancelled: Job aborted due to stage failure: Task 0 in stage 0.0 failed 4 times, most recent failure: Lost task 0.3 in stage 0.0 (TID 3) (172.18.0.7 executor 0): org.apache.spark.SparkFileNotFoundException: File file:/tmp/spark-d647b6e1-10af-46db-9915-183687f7eab0/userFiles-b0c4d8f8-2b5b-44b3-8f66-ae1674e6840f/test1234.csv does not exist
[2024-12-08T21:40:30.424+0000] {spark_submit.py:634} INFO - It is possible the underlying files have been updated. You can explicitly invalidate the cache in Spark by running 'REFRESH TABLE tableName' command in SQL or by recreating the Dataset/DataFrame involved.
[2024-12-08T21:40:30.424+0000] {spark_submit.py:634} INFO - at org.apache.spark.sql.errors.QueryExecutionErrors$.readCurrentFileNotFoundError(QueryExecutionErrors.scala:781)
[2024-12-08T21:40:30.424+0000] {spark_submit.py:634} INFO - at org.apache.spark.sql.execution.datasources.FileScanRDD$$anon$1.org$apache$spark$sql$execution$datasources$FileScanRDD$$anon$$readCurrentFile(FileScanRDD.scala:220)
[2024-12-08T21:40:30.424+0000] {spark_submit.py:634} INFO - at org.apache.spark.sql.execution.datasources.FileScanRDD$$anon$1.nextIterator(FileScanRDD.scala:279)
[2024-12-08T21:40:30.424+0000] {spark_submit.py:634} INFO - at org.apache.spark.sql.execution.datasources.FileScanRDD$$anon$1.hasNext(FileScanRDD.scala:129)
[2024-12-08T21:40:30.425+0000] {spark_submit.py:634} INFO - at scala.collection.Iterator$$anon$10.hasNext(Iterator.scala:460)
[2024-12-08T21:40:30.425+0000] {spark_submit.py:634} INFO - at org.apache.spark.sql.catalyst.expressions.GeneratedClass$GeneratedIteratorForCodegenStage1.processNext(Unknown Source)
[2024-12-08T21:40:30.425+0000] {spark_submit.py:634} INFO - at org.apache.spark.sql.execution.BufferedRowIterator.hasNext(BufferedRowIterator.java:43)
[2024-12-08T21:40:30.426+0000] {spark_submit.py:634} INFO - at org.apache.spark.sql.execution.WholeStageCodegenEvaluatorFactory$WholeStageCodegenPartitionEvaluator$$anon$1.hasNext(WholeStageCodegenEvaluatorFactory.scala:43)
[2024-12-08T21:40:30.426+0000] {spark_submit.py:634} INFO - at org.apache.spark.sql.execution.SparkPlan.$anonfun$getByteArrayRdd$1(SparkPlan.scala:388)
[2024-12-08T21:40:30.427+0000] {spark_submit.py:634} INFO - at org.apache.spark.rdd.RDD.$anonfun$mapPartitionsInternal$2(RDD.scala:893)
[2024-12-08T21:40:30.427+0000] {spark_submit.py:634} INFO - at org.apache.spark.rdd.RDD.$anonfun$mapPartitionsInternal$2$adapted(RDD.scala:893)
[2024-12-08T21:40:30.427+0000] {spark_submit.py:634} INFO - at org.apache.spark.rdd.MapPartitionsRDD.compute(MapPartitionsRDD.scala:52)
[2024-12-08T21:40:30.427+0000] {spark_submit.py:634} INFO - at org.apache.spark.rdd.RDD.computeOrReadCheckpoint(RDD.scala:367)
[2024-12-08T21:40:30.427+0000] {spark_submit.py:634} INFO - at org.apache.spark.rdd.RDD.iterator(RDD.scala:331)
[2024-12-08T21:40:30.428+0000] {spark_submit.py:634} INFO - at org.apache.spark.scheduler.ResultTask.runTask(ResultTask.scala:93)
[2024-12-08T21:40:30.428+0000] {spark_submit.py:634} INFO - at org.apache.spark.TaskContext.runTaskWithListeners(TaskContext.scala:166)
[2024-12-08T21:40:30.428+0000] {spark_submit.py:634} INFO - at org.apache.spark.scheduler.Task.run(Task.scala:141)
[2024-12-08T21:40:30.428+0000] {spark_submit.py:634} INFO - at org.apache.spark.executor.Executor$TaskRunner.$anonfun$run$4(Executor.scala:620)
[2024-12-08T21:40:30.428+0000] {spark_submit.py:634} INFO - at org.apache.spark.util.SparkErrorUtils.tryWithSafeFinally(SparkErrorUtils.scala:64)
[2024-12-08T21:40:30.428+0000] {spark_submit.py:634} INFO - at org.apache.spark.util.SparkErrorUtils.tryWithSafeFinally$(SparkErrorUtils.scala:61)
[2024-12-08T21:40:30.429+0000] {spark_submit.py:634} INFO - at org.apache.spark.util.Utils$.tryWithSafeFinally(Utils.scala:94)
[2024-12-08T21:40:30.429+0000] {spark_submit.py:634} INFO - at org.apache.spark.executor.Executor$TaskRunner.run(Executor.scala:623)
[2024-12-08T21:40:30.429+0000] {spark_submit.py:634} INFO - at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
[2024-12-08T21:40:30.429+0000] {spark_submit.py:634} INFO - at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
[2024-12-08T21:40:30.430+0000] {spark_submit.py:634} INFO - at java.base/java.lang.Thread.run(Thread.java:840)
[2024-12-08T21:40:30.430+0000] {spark_submit.py:634} INFO - 
[2024-12-08T21:40:30.430+0000] {spark_submit.py:634} INFO - Driver stacktrace:
[2024-12-08T21:40:30.430+0000] {spark_submit.py:634} INFO - 24/12/08 21:40:30 INFO DAGScheduler: ResultStage 0 (csv at NativeMethodAccessorImpl.java:0) failed in 2.522 s due to Job aborted due to stage failure: Task 0 in stage 0.0 failed 4 times, most recent failure: Lost task 0.3 in stage 0.0 (TID 3) (172.18.0.7 executor 0): org.apache.spark.SparkFileNotFoundException: File file:/tmp/spark-d647b6e1-10af-46db-9915-183687f7eab0/userFiles-b0c4d8f8-2b5b-44b3-8f66-ae1674e6840f/test1234.csv does not exist
[2024-12-08T21:40:30.431+0000] {spark_submit.py:634} INFO - It is possible the underlying files have been updated. You can explicitly invalidate the cache in Spark by running 'REFRESH TABLE tableName' command in SQL or by recreating the Dataset/DataFrame involved.
[2024-12-08T21:40:30.431+0000] {spark_submit.py:634} INFO - at org.apache.spark.sql.errors.QueryExecutionErrors$.readCurrentFileNotFoundError(QueryExecutionErrors.scala:781)
[2024-12-08T21:40:30.431+0000] {spark_submit.py:634} INFO - at org.apache.spark.sql.execution.datasources.FileScanRDD$$anon$1.org$apache$spark$sql$execution$datasources$FileScanRDD$$anon$$readCurrentFile(FileScanRDD.scala:220)
[2024-12-08T21:40:30.431+0000] {spark_submit.py:634} INFO - at org.apache.spark.sql.execution.datasources.FileScanRDD$$anon$1.nextIterator(FileScanRDD.scala:279)
[2024-12-08T21:40:30.431+0000] {spark_submit.py:634} INFO - at org.apache.spark.sql.execution.datasources.FileScanRDD$$anon$1.hasNext(FileScanRDD.scala:129)
[2024-12-08T21:40:30.432+0000] {spark_submit.py:634} INFO - at scala.collection.Iterator$$anon$10.hasNext(Iterator.scala:460)
[2024-12-08T21:40:30.432+0000] {spark_submit.py:634} INFO - at org.apache.spark.sql.catalyst.expressions.GeneratedClass$GeneratedIteratorForCodegenStage1.processNext(Unknown Source)
[2024-12-08T21:40:30.432+0000] {spark_submit.py:634} INFO - at org.apache.spark.sql.execution.BufferedRowIterator.hasNext(BufferedRowIterator.java:43)
[2024-12-08T21:40:30.432+0000] {spark_submit.py:634} INFO - at org.apache.spark.sql.execution.WholeStageCodegenEvaluatorFactory$WholeStageCodegenPartitionEvaluator$$anon$1.hasNext(WholeStageCodegenEvaluatorFactory.scala:43)
[2024-12-08T21:40:30.432+0000] {spark_submit.py:634} INFO - at org.apache.spark.sql.execution.SparkPlan.$anonfun$getByteArrayRdd$1(SparkPlan.scala:388)
[2024-12-08T21:40:30.433+0000] {spark_submit.py:634} INFO - at org.apache.spark.rdd.RDD.$anonfun$mapPartitionsInternal$2(RDD.scala:893)
[2024-12-08T21:40:30.433+0000] {spark_submit.py:634} INFO - at org.apache.spark.rdd.RDD.$anonfun$mapPartitionsInternal$2$adapted(RDD.scala:893)
[2024-12-08T21:40:30.433+0000] {spark_submit.py:634} INFO - at org.apache.spark.rdd.MapPartitionsRDD.compute(MapPartitionsRDD.scala:52)
[2024-12-08T21:40:30.433+0000] {spark_submit.py:634} INFO - at org.apache.spark.rdd.RDD.computeOrReadCheckpoint(RDD.scala:367)
[2024-12-08T21:40:30.433+0000] {spark_submit.py:634} INFO - at org.apache.spark.rdd.RDD.iterator(RDD.scala:331)
[2024-12-08T21:40:30.434+0000] {spark_submit.py:634} INFO - at org.apache.spark.scheduler.ResultTask.runTask(ResultTask.scala:93)
[2024-12-08T21:40:30.434+0000] {spark_submit.py:634} INFO - at org.apache.spark.TaskContext.runTaskWithListeners(TaskContext.scala:166)
[2024-12-08T21:40:30.434+0000] {spark_submit.py:634} INFO - at org.apache.spark.scheduler.Task.run(Task.scala:141)
[2024-12-08T21:40:30.434+0000] {spark_submit.py:634} INFO - at org.apache.spark.executor.Executor$TaskRunner.$anonfun$run$4(Executor.scala:620)
[2024-12-08T21:40:30.434+0000] {spark_submit.py:634} INFO - at org.apache.spark.util.SparkErrorUtils.tryWithSafeFinally(SparkErrorUtils.scala:64)
[2024-12-08T21:40:30.434+0000] {spark_submit.py:634} INFO - at org.apache.spark.util.SparkErrorUtils.tryWithSafeFinally$(SparkErrorUtils.scala:61)
[2024-12-08T21:40:30.434+0000] {spark_submit.py:634} INFO - at org.apache.spark.util.Utils$.tryWithSafeFinally(Utils.scala:94)
[2024-12-08T21:40:30.434+0000] {spark_submit.py:634} INFO - at org.apache.spark.executor.Executor$TaskRunner.run(Executor.scala:623)
[2024-12-08T21:40:30.435+0000] {spark_submit.py:634} INFO - at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
[2024-12-08T21:40:30.435+0000] {spark_submit.py:634} INFO - at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
[2024-12-08T21:40:30.435+0000] {spark_submit.py:634} INFO - at java.base/java.lang.Thread.run(Thread.java:840)
[2024-12-08T21:40:30.435+0000] {spark_submit.py:634} INFO - 
[2024-12-08T21:40:30.435+0000] {spark_submit.py:634} INFO - Driver stacktrace:
[2024-12-08T21:40:30.435+0000] {spark_submit.py:634} INFO - 24/12/08 21:40:30 INFO DAGScheduler: Job 0 failed: csv at NativeMethodAccessorImpl.java:0, took 2.564855 s
[2024-12-08T21:40:30.606+0000] {spark_submit.py:634} INFO - Traceback (most recent call last):
[2024-12-08T21:40:30.607+0000] {spark_submit.py:634} INFO - File "/opt/***/jobs/python/wordcountjob.py", line 32, in <module>
[2024-12-08T21:40:30.608+0000] {spark_submit.py:634} INFO - df = spark.read.csv(path=myfile, header=True, inferSchema=True)
[2024-12-08T21:40:30.608+0000] {spark_submit.py:634} INFO - ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
[2024-12-08T21:40:30.608+0000] {spark_submit.py:634} INFO - File "/home/***/.local/lib/python3.12/site-packages/pyspark/python/lib/pyspark.zip/pyspark/sql/readwriter.py", line 740, in csv
[2024-12-08T21:40:30.609+0000] {spark_submit.py:634} INFO - File "/home/***/.local/lib/python3.12/site-packages/pyspark/python/lib/py4j-0.10.9.7-src.zip/py4j/java_gateway.py", line 1322, in __call__
[2024-12-08T21:40:30.609+0000] {spark_submit.py:634} INFO - File "/home/***/.local/lib/python3.12/site-packages/pyspark/python/lib/pyspark.zip/pyspark/errors/exceptions/captured.py", line 179, in deco
[2024-12-08T21:40:30.610+0000] {spark_submit.py:634} INFO - File "/home/***/.local/lib/python3.12/site-packages/pyspark/python/lib/py4j-0.10.9.7-src.zip/py4j/protocol.py", line 326, in get_return_value
[2024-12-08T21:40:30.653+0000] {spark_submit.py:634} INFO - py4j.protocol.Py4JJavaError: An error occurred while calling o27.csv.
[2024-12-08T21:40:30.657+0000] {spark_submit.py:634} INFO - : org.apache.spark.SparkException: Job aborted due to stage failure: Task 0 in stage 0.0 failed 4 times, most recent failure: Lost task 0.3 in stage 0.0 (TID 3) (172.18.0.7 executor 0): org.apache.spark.SparkFileNotFoundException: File file:/tmp/spark-d647b6e1-10af-46db-9915-183687f7eab0/userFiles-b0c4d8f8-2b5b-44b3-8f66-ae1674e6840f/test1234.csv does not exist
[2024-12-08T21:40:30.658+0000] {spark_submit.py:634} INFO - It is possible the underlying files have been updated. You can explicitly invalidate the cache in Spark by running 'REFRESH TABLE tableName' command in SQL or by recreating the Dataset/DataFrame involved.
[2024-12-08T21:40:30.659+0000] {spark_submit.py:634} INFO - at org.apache.spark.sql.errors.QueryExecutionErrors$.readCurrentFileNotFoundError(QueryExecutionErrors.scala:781)
[2024-12-08T21:40:30.661+0000] {spark_submit.py:634} INFO - at org.apache.spark.sql.execution.datasources.FileScanRDD$$anon$1.org$apache$spark$sql$execution$datasources$FileScanRDD$$anon$$readCurrentFile(FileScanRDD.scala:220)
[2024-12-08T21:40:30.662+0000] {spark_submit.py:634} INFO - at org.apache.spark.sql.execution.datasources.FileScanRDD$$anon$1.nextIterator(FileScanRDD.scala:279)
[2024-12-08T21:40:30.662+0000] {spark_submit.py:634} INFO - at org.apache.spark.sql.execution.datasources.FileScanRDD$$anon$1.hasNext(FileScanRDD.scala:129)
[2024-12-08T21:40:30.662+0000] {spark_submit.py:634} INFO - at scala.collection.Iterator$$anon$10.hasNext(Iterator.scala:460)
[2024-12-08T21:40:30.662+0000] {spark_submit.py:634} INFO - at org.apache.spark.sql.catalyst.expressions.GeneratedClass$GeneratedIteratorForCodegenStage1.processNext(Unknown Source)
[2024-12-08T21:40:30.663+0000] {spark_submit.py:634} INFO - at org.apache.spark.sql.execution.BufferedRowIterator.hasNext(BufferedRowIterator.java:43)
[2024-12-08T21:40:30.666+0000] {spark_submit.py:634} INFO - at org.apache.spark.sql.execution.WholeStageCodegenEvaluatorFactory$WholeStageCodegenPartitionEvaluator$$anon$1.hasNext(WholeStageCodegenEvaluatorFactory.scala:43)
[2024-12-08T21:40:30.666+0000] {spark_submit.py:634} INFO - at org.apache.spark.sql.execution.SparkPlan.$anonfun$getByteArrayRdd$1(SparkPlan.scala:388)
[2024-12-08T21:40:30.666+0000] {spark_submit.py:634} INFO - at org.apache.spark.rdd.RDD.$anonfun$mapPartitionsInternal$2(RDD.scala:893)
[2024-12-08T21:40:30.666+0000] {spark_submit.py:634} INFO - at org.apache.spark.rdd.RDD.$anonfun$mapPartitionsInternal$2$adapted(RDD.scala:893)
[2024-12-08T21:40:30.667+0000] {spark_submit.py:634} INFO - at org.apache.spark.rdd.MapPartitionsRDD.compute(MapPartitionsRDD.scala:52)
[2024-12-08T21:40:30.667+0000] {spark_submit.py:634} INFO - at org.apache.spark.rdd.RDD.computeOrReadCheckpoint(RDD.scala:367)
[2024-12-08T21:40:30.667+0000] {spark_submit.py:634} INFO - at org.apache.spark.rdd.RDD.iterator(RDD.scala:331)
[2024-12-08T21:40:30.667+0000] {spark_submit.py:634} INFO - at org.apache.spark.scheduler.ResultTask.runTask(ResultTask.scala:93)
[2024-12-08T21:40:30.667+0000] {spark_submit.py:634} INFO - at org.apache.spark.TaskContext.runTaskWithListeners(TaskContext.scala:166)
[2024-12-08T21:40:30.667+0000] {spark_submit.py:634} INFO - at org.apache.spark.scheduler.Task.run(Task.scala:141)
[2024-12-08T21:40:30.668+0000] {spark_submit.py:634} INFO - at org.apache.spark.executor.Executor$TaskRunner.$anonfun$run$4(Executor.scala:620)
[2024-12-08T21:40:30.668+0000] {spark_submit.py:634} INFO - at org.apache.spark.util.SparkErrorUtils.tryWithSafeFinally(SparkErrorUtils.scala:64)
[2024-12-08T21:40:30.668+0000] {spark_submit.py:634} INFO - at org.apache.spark.util.SparkErrorUtils.tryWithSafeFinally$(SparkErrorUtils.scala:61)
[2024-12-08T21:40:30.668+0000] {spark_submit.py:634} INFO - at org.apache.spark.util.Utils$.tryWithSafeFinally(Utils.scala:94)
[2024-12-08T21:40:30.668+0000] {spark_submit.py:634} INFO - at org.apache.spark.executor.Executor$TaskRunner.run(Executor.scala:623)
[2024-12-08T21:40:30.668+0000] {spark_submit.py:634} INFO - at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
[2024-12-08T21:40:30.668+0000] {spark_submit.py:634} INFO - at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
[2024-12-08T21:40:30.669+0000] {spark_submit.py:634} INFO - at java.base/java.lang.Thread.run(Thread.java:840)
[2024-12-08T21:40:30.669+0000] {spark_submit.py:634} INFO - 
[2024-12-08T21:40:30.669+0000] {spark_submit.py:634} INFO - Driver stacktrace:
[2024-12-08T21:40:30.669+0000] {spark_submit.py:634} INFO - at org.apache.spark.scheduler.DAGScheduler.failJobAndIndependentStages(DAGScheduler.scala:2856)
[2024-12-08T21:40:30.669+0000] {spark_submit.py:634} INFO - at org.apache.spark.scheduler.DAGScheduler.$anonfun$abortStage$2(DAGScheduler.scala:2792)
[2024-12-08T21:40:30.669+0000] {spark_submit.py:634} INFO - at org.apache.spark.scheduler.DAGScheduler.$anonfun$abortStage$2$adapted(DAGScheduler.scala:2791)
[2024-12-08T21:40:30.670+0000] {spark_submit.py:634} INFO - at scala.collection.mutable.ResizableArray.foreach(ResizableArray.scala:62)
[2024-12-08T21:40:30.670+0000] {spark_submit.py:634} INFO - at scala.collection.mutable.ResizableArray.foreach$(ResizableArray.scala:55)
[2024-12-08T21:40:30.670+0000] {spark_submit.py:634} INFO - at scala.collection.mutable.ArrayBuffer.foreach(ArrayBuffer.scala:49)
[2024-12-08T21:40:30.670+0000] {spark_submit.py:634} INFO - at org.apache.spark.scheduler.DAGScheduler.abortStage(DAGScheduler.scala:2791)
[2024-12-08T21:40:30.670+0000] {spark_submit.py:634} INFO - at org.apache.spark.scheduler.DAGScheduler.$anonfun$handleTaskSetFailed$1(DAGScheduler.scala:1247)
[2024-12-08T21:40:30.671+0000] {spark_submit.py:634} INFO - at org.apache.spark.scheduler.DAGScheduler.$anonfun$handleTaskSetFailed$1$adapted(DAGScheduler.scala:1247)
[2024-12-08T21:40:30.673+0000] {spark_submit.py:634} INFO - at scala.Option.foreach(Option.scala:407)
[2024-12-08T21:40:30.673+0000] {spark_submit.py:634} INFO - at org.apache.spark.scheduler.DAGScheduler.handleTaskSetFailed(DAGScheduler.scala:1247)
[2024-12-08T21:40:30.673+0000] {spark_submit.py:634} INFO - at org.apache.spark.scheduler.DAGSchedulerEventProcessLoop.doOnReceive(DAGScheduler.scala:3060)
[2024-12-08T21:40:30.673+0000] {spark_submit.py:634} INFO - at org.apache.spark.scheduler.DAGSchedulerEventProcessLoop.onReceive(DAGScheduler.scala:2994)
[2024-12-08T21:40:30.674+0000] {spark_submit.py:634} INFO - at org.apache.spark.scheduler.DAGSchedulerEventProcessLoop.onReceive(DAGScheduler.scala:2983)
[2024-12-08T21:40:30.674+0000] {spark_submit.py:634} INFO - at org.apache.spark.util.EventLoop$$anon$1.run(EventLoop.scala:49)
[2024-12-08T21:40:30.674+0000] {spark_submit.py:634} INFO - at org.apache.spark.scheduler.DAGScheduler.runJob(DAGScheduler.scala:989)
[2024-12-08T21:40:30.674+0000] {spark_submit.py:634} INFO - at org.apache.spark.SparkContext.runJob(SparkContext.scala:2393)
[2024-12-08T21:40:30.674+0000] {spark_submit.py:634} INFO - at org.apache.spark.SparkContext.runJob(SparkContext.scala:2414)
[2024-12-08T21:40:30.674+0000] {spark_submit.py:634} INFO - at org.apache.spark.SparkContext.runJob(SparkContext.scala:2433)
[2024-12-08T21:40:30.674+0000] {spark_submit.py:634} INFO - at org.apache.spark.sql.execution.SparkPlan.executeTake(SparkPlan.scala:530)
[2024-12-08T21:40:30.675+0000] {spark_submit.py:634} INFO - at org.apache.spark.sql.execution.SparkPlan.executeTake(SparkPlan.scala:483)
[2024-12-08T21:40:30.675+0000] {spark_submit.py:634} INFO - at org.apache.spark.sql.execution.CollectLimitExec.executeCollect(limit.scala:61)
[2024-12-08T21:40:30.675+0000] {spark_submit.py:634} INFO - at org.apache.spark.sql.Dataset.collectFromPlan(Dataset.scala:4333)
[2024-12-08T21:40:30.675+0000] {spark_submit.py:634} INFO - at org.apache.spark.sql.Dataset.$anonfun$head$1(Dataset.scala:3316)
[2024-12-08T21:40:30.675+0000] {spark_submit.py:634} INFO - at org.apache.spark.sql.Dataset.$anonfun$withAction$2(Dataset.scala:4323)
[2024-12-08T21:40:30.675+0000] {spark_submit.py:634} INFO - at org.apache.spark.sql.execution.QueryExecution$.withInternalError(QueryExecution.scala:546)
[2024-12-08T21:40:30.677+0000] {spark_submit.py:634} INFO - at org.apache.spark.sql.Dataset.$anonfun$withAction$1(Dataset.scala:4321)
[2024-12-08T21:40:30.678+0000] {spark_submit.py:634} INFO - at org.apache.spark.sql.execution.SQLExecution$.$anonfun$withNewExecutionId$6(SQLExecution.scala:125)
[2024-12-08T21:40:30.678+0000] {spark_submit.py:634} INFO - at org.apache.spark.sql.execution.SQLExecution$.withSQLConfPropagated(SQLExecution.scala:201)
[2024-12-08T21:40:30.678+0000] {spark_submit.py:634} INFO - at org.apache.spark.sql.execution.SQLExecution$.$anonfun$withNewExecutionId$1(SQLExecution.scala:108)
[2024-12-08T21:40:30.678+0000] {spark_submit.py:634} INFO - at org.apache.spark.sql.SparkSession.withActive(SparkSession.scala:900)
[2024-12-08T21:40:30.678+0000] {spark_submit.py:634} INFO - at org.apache.spark.sql.execution.SQLExecution$.withNewExecutionId(SQLExecution.scala:66)
[2024-12-08T21:40:30.679+0000] {spark_submit.py:634} INFO - at org.apache.spark.sql.Dataset.withAction(Dataset.scala:4321)
[2024-12-08T21:40:30.679+0000] {spark_submit.py:634} INFO - at org.apache.spark.sql.Dataset.head(Dataset.scala:3316)
[2024-12-08T21:40:30.679+0000] {spark_submit.py:634} INFO - at org.apache.spark.sql.Dataset.take(Dataset.scala:3539)
[2024-12-08T21:40:30.679+0000] {spark_submit.py:634} INFO - at org.apache.spark.sql.execution.datasources.csv.TextInputCSVDataSource$.infer(CSVDataSource.scala:111)
[2024-12-08T21:40:30.680+0000] {spark_submit.py:634} INFO - at org.apache.spark.sql.execution.datasources.csv.CSVDataSource.inferSchema(CSVDataSource.scala:64)
[2024-12-08T21:40:30.680+0000] {spark_submit.py:634} INFO - at org.apache.spark.sql.execution.datasources.csv.CSVFileFormat.inferSchema(CSVFileFormat.scala:62)
[2024-12-08T21:40:30.680+0000] {spark_submit.py:634} INFO - at org.apache.spark.sql.execution.datasources.DataSource.$anonfun$getOrInferFileFormatSchema$11(DataSource.scala:208)
[2024-12-08T21:40:30.680+0000] {spark_submit.py:634} INFO - at scala.Option.orElse(Option.scala:447)
[2024-12-08T21:40:30.680+0000] {spark_submit.py:634} INFO - at org.apache.spark.sql.execution.datasources.DataSource.getOrInferFileFormatSchema(DataSource.scala:205)
[2024-12-08T21:40:30.681+0000] {spark_submit.py:634} INFO - at org.apache.spark.sql.execution.datasources.DataSource.resolveRelation(DataSource.scala:407)
[2024-12-08T21:40:30.681+0000] {spark_submit.py:634} INFO - at org.apache.spark.sql.DataFrameReader.loadV1Source(DataFrameReader.scala:229)
[2024-12-08T21:40:30.681+0000] {spark_submit.py:634} INFO - at org.apache.spark.sql.DataFrameReader.$anonfun$load$2(DataFrameReader.scala:211)
[2024-12-08T21:40:30.681+0000] {spark_submit.py:634} INFO - at scala.Option.getOrElse(Option.scala:189)
[2024-12-08T21:40:30.681+0000] {spark_submit.py:634} INFO - at org.apache.spark.sql.DataFrameReader.load(DataFrameReader.scala:211)
[2024-12-08T21:40:30.681+0000] {spark_submit.py:634} INFO - at org.apache.spark.sql.DataFrameReader.csv(DataFrameReader.scala:538)
[2024-12-08T21:40:30.682+0000] {spark_submit.py:634} INFO - at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
[2024-12-08T21:40:30.683+0000] {spark_submit.py:634} INFO - at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
[2024-12-08T21:40:30.683+0000] {spark_submit.py:634} INFO - at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
[2024-12-08T21:40:30.683+0000] {spark_submit.py:634} INFO - at java.base/java.lang.reflect.Method.invoke(Method.java:569)
[2024-12-08T21:40:30.683+0000] {spark_submit.py:634} INFO - at py4j.reflection.MethodInvoker.invoke(MethodInvoker.java:244)
[2024-12-08T21:40:30.683+0000] {spark_submit.py:634} INFO - at py4j.reflection.ReflectionEngine.invoke(ReflectionEngine.java:374)
[2024-12-08T21:40:30.684+0000] {spark_submit.py:634} INFO - at py4j.Gateway.invoke(Gateway.java:282)
[2024-12-08T21:40:30.684+0000] {spark_submit.py:634} INFO - at py4j.commands.AbstractCommand.invokeMethod(AbstractCommand.java:132)
[2024-12-08T21:40:30.684+0000] {spark_submit.py:634} INFO - at py4j.commands.CallCommand.execute(CallCommand.java:79)
[2024-12-08T21:40:30.685+0000] {spark_submit.py:634} INFO - at py4j.ClientServerConnection.waitForCommands(ClientServerConnection.java:182)
[2024-12-08T21:40:30.686+0000] {spark_submit.py:634} INFO - at py4j.ClientServerConnection.run(ClientServerConnection.java:106)
[2024-12-08T21:40:30.687+0000] {spark_submit.py:634} INFO - at java.base/java.lang.Thread.run(Thread.java:840)
[2024-12-08T21:40:30.687+0000] {spark_submit.py:634} INFO - Caused by: org.apache.spark.SparkFileNotFoundException: File file:/tmp/spark-d647b6e1-10af-46db-9915-183687f7eab0/userFiles-b0c4d8f8-2b5b-44b3-8f66-ae1674e6840f/test1234.csv does not exist
[2024-12-08T21:40:30.687+0000] {spark_submit.py:634} INFO - It is possible the underlying files have been updated. You can explicitly invalidate the cache in Spark by running 'REFRESH TABLE tableName' command in SQL or by recreating the Dataset/DataFrame involved.
[2024-12-08T21:40:30.688+0000] {spark_submit.py:634} INFO - at org.apache.spark.sql.errors.QueryExecutionErrors$.readCurrentFileNotFoundError(QueryExecutionErrors.scala:781)
[2024-12-08T21:40:30.689+0000] {spark_submit.py:634} INFO - at org.apache.spark.sql.execution.datasources.FileScanRDD$$anon$1.org$apache$spark$sql$execution$datasources$FileScanRDD$$anon$$readCurrentFile(FileScanRDD.scala:220)
[2024-12-08T21:40:30.690+0000] {spark_submit.py:634} INFO - at org.apache.spark.sql.execution.datasources.FileScanRDD$$anon$1.nextIterator(FileScanRDD.scala:279)
[2024-12-08T21:40:30.691+0000] {spark_submit.py:634} INFO - at org.apache.spark.sql.execution.datasources.FileScanRDD$$anon$1.hasNext(FileScanRDD.scala:129)
[2024-12-08T21:40:30.691+0000] {spark_submit.py:634} INFO - at scala.collection.Iterator$$anon$10.hasNext(Iterator.scala:460)
[2024-12-08T21:40:30.691+0000] {spark_submit.py:634} INFO - at org.apache.spark.sql.catalyst.expressions.GeneratedClass$GeneratedIteratorForCodegenStage1.processNext(Unknown Source)
[2024-12-08T21:40:30.691+0000] {spark_submit.py:634} INFO - at org.apache.spark.sql.execution.BufferedRowIterator.hasNext(BufferedRowIterator.java:43)
[2024-12-08T21:40:30.691+0000] {spark_submit.py:634} INFO - at org.apache.spark.sql.execution.WholeStageCodegenEvaluatorFactory$WholeStageCodegenPartitionEvaluator$$anon$1.hasNext(WholeStageCodegenEvaluatorFactory.scala:43)
[2024-12-08T21:40:30.692+0000] {spark_submit.py:634} INFO - at org.apache.spark.sql.execution.SparkPlan.$anonfun$getByteArrayRdd$1(SparkPlan.scala:388)
[2024-12-08T21:40:30.692+0000] {spark_submit.py:634} INFO - at org.apache.spark.rdd.RDD.$anonfun$mapPartitionsInternal$2(RDD.scala:893)
[2024-12-08T21:40:30.692+0000] {spark_submit.py:634} INFO - at org.apache.spark.rdd.RDD.$anonfun$mapPartitionsInternal$2$adapted(RDD.scala:893)
[2024-12-08T21:40:30.692+0000] {spark_submit.py:634} INFO - at org.apache.spark.rdd.MapPartitionsRDD.compute(MapPartitionsRDD.scala:52)
[2024-12-08T21:40:30.693+0000] {spark_submit.py:634} INFO - at org.apache.spark.rdd.RDD.computeOrReadCheckpoint(RDD.scala:367)
[2024-12-08T21:40:30.695+0000] {spark_submit.py:634} INFO - at org.apache.spark.rdd.RDD.iterator(RDD.scala:331)
[2024-12-08T21:40:30.695+0000] {spark_submit.py:634} INFO - at org.apache.spark.scheduler.ResultTask.runTask(ResultTask.scala:93)
[2024-12-08T21:40:30.695+0000] {spark_submit.py:634} INFO - at org.apache.spark.TaskContext.runTaskWithListeners(TaskContext.scala:166)
[2024-12-08T21:40:30.695+0000] {spark_submit.py:634} INFO - at org.apache.spark.scheduler.Task.run(Task.scala:141)
[2024-12-08T21:40:30.696+0000] {spark_submit.py:634} INFO - at org.apache.spark.executor.Executor$TaskRunner.$anonfun$run$4(Executor.scala:620)
[2024-12-08T21:40:30.696+0000] {spark_submit.py:634} INFO - at org.apache.spark.util.SparkErrorUtils.tryWithSafeFinally(SparkErrorUtils.scala:64)
[2024-12-08T21:40:30.696+0000] {spark_submit.py:634} INFO - at org.apache.spark.util.SparkErrorUtils.tryWithSafeFinally$(SparkErrorUtils.scala:61)
[2024-12-08T21:40:30.696+0000] {spark_submit.py:634} INFO - at org.apache.spark.util.Utils$.tryWithSafeFinally(Utils.scala:94)
[2024-12-08T21:40:30.696+0000] {spark_submit.py:634} INFO - at org.apache.spark.executor.Executor$TaskRunner.run(Executor.scala:623)
[2024-12-08T21:40:30.696+0000] {spark_submit.py:634} INFO - at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
[2024-12-08T21:40:30.696+0000] {spark_submit.py:634} INFO - at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
[2024-12-08T21:40:30.696+0000] {spark_submit.py:634} INFO - ... 1 more
[2024-12-08T21:40:30.696+0000] {spark_submit.py:634} INFO - 
[2024-12-08T21:40:30.740+0000] {spark_submit.py:634} INFO - 24/12/08 21:40:30 INFO SparkContext: Invoking stop() from shutdown hook
[2024-12-08T21:40:30.740+0000] {spark_submit.py:634} INFO - 24/12/08 21:40:30 INFO SparkContext: SparkContext is stopping with exitCode 0.
[2024-12-08T21:40:30.748+0000] {spark_submit.py:634} INFO - 24/12/08 21:40:30 INFO SparkUI: Stopped Spark web UI at http://2a892a3469ec:4042
[2024-12-08T21:40:30.750+0000] {spark_submit.py:634} INFO - 24/12/08 21:40:30 INFO StandaloneSchedulerBackend: Shutting down all executors
[2024-12-08T21:40:30.751+0000] {spark_submit.py:634} INFO - 24/12/08 21:40:30 INFO StandaloneSchedulerBackend$StandaloneDriverEndpoint: Asking each executor to shut down
[2024-12-08T21:40:30.772+0000] {spark_submit.py:634} INFO - 24/12/08 21:40:30 INFO MapOutputTrackerMasterEndpoint: MapOutputTrackerMasterEndpoint stopped!
[2024-12-08T21:40:30.803+0000] {spark_submit.py:634} INFO - 24/12/08 21:40:30 INFO MemoryStore: MemoryStore cleared
[2024-12-08T21:40:30.804+0000] {spark_submit.py:634} INFO - 24/12/08 21:40:30 INFO BlockManager: BlockManager stopped
[2024-12-08T21:40:30.811+0000] {spark_submit.py:634} INFO - 24/12/08 21:40:30 INFO BlockManagerMaster: BlockManagerMaster stopped
[2024-12-08T21:40:30.815+0000] {spark_submit.py:634} INFO - 24/12/08 21:40:30 INFO OutputCommitCoordinator$OutputCommitCoordinatorEndpoint: OutputCommitCoordinator stopped!
[2024-12-08T21:40:30.824+0000] {spark_submit.py:634} INFO - 24/12/08 21:40:30 INFO SparkContext: Successfully stopped SparkContext
[2024-12-08T21:40:30.824+0000] {spark_submit.py:634} INFO - 24/12/08 21:40:30 INFO ShutdownHookManager: Shutdown hook called
[2024-12-08T21:40:30.825+0000] {spark_submit.py:634} INFO - 24/12/08 21:40:30 INFO ShutdownHookManager: Deleting directory /tmp/spark-d647b6e1-10af-46db-9915-183687f7eab0/pyspark-72097a14-c55f-4a27-be75-b5fcc78fe50a
[2024-12-08T21:40:30.831+0000] {spark_submit.py:634} INFO - 24/12/08 21:40:30 INFO ShutdownHookManager: Deleting directory /tmp/spark-ba651176-c3bc-4b95-a82d-869109818533
[2024-12-08T21:40:30.836+0000] {spark_submit.py:634} INFO - 24/12/08 21:40:30 INFO ShutdownHookManager: Deleting directory /tmp/spark-d647b6e1-10af-46db-9915-183687f7eab0
[2024-12-08T21:40:30.904+0000] {taskinstance.py:3311} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 767, in _execute_task
    result = _execute_callable(context=context, **execute_callable_kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 733, in _execute_callable
    return ExecutionCallableRunner(
           ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/utils/operator_helpers.py", line 252, in run
    return self.func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/baseoperator.py", line 417, in wrapper
    return func(self, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/providers/apache/spark/operators/spark_submit.py", line 176, in execute
    self._hook.submit(self.application)
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/providers/apache/spark/hooks/spark_submit.py", line 560, in submit
    raise AirflowException(
airflow.exceptions.AirflowException: Cannot execute: spark-submit --master spark://sparkingflow-spark-master-1:7077 --name arrow-spark --deploy-mode client jobs/python/wordcountjob.py. Error code is: 1.
[2024-12-08T21:40:30.917+0000] {taskinstance.py:1225} INFO - Marking task as FAILED. dag_id=sparking_flow, task_id=python_job, run_id=manual__2024-12-08T21:40:11.545590+00:00, execution_date=20241208T214011, start_date=20241208T214013, end_date=20241208T214030
[2024-12-08T21:40:30.940+0000] {taskinstance.py:340} INFO - ::group::Post task execution logs
[2024-12-08T21:40:30.941+0000] {standard_task_runner.py:124} ERROR - Failed to execute job 271 for task python_job (Cannot execute: spark-submit --master spark://sparkingflow-spark-master-1:7077 --name arrow-spark --deploy-mode client jobs/python/wordcountjob.py. Error code is: 1.; 189)
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/task/task_runner/standard_task_runner.py", line 117, in _start_by_fork
    ret = args.func(args, dag=self.dag)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/cli/cli_config.py", line 49, in command
    return func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/utils/cli.py", line 116, in wrapper
    return f(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/cli/commands/task_command.py", line 483, in task_run
    task_return_code = _run_task_by_selected_method(args, _dag, ti)
                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/cli/commands/task_command.py", line 256, in _run_task_by_selected_method
    return _run_raw_task(args, ti)
           ^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/cli/commands/task_command.py", line 341, in _run_raw_task
    return ti._run_raw_task(
           ^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/utils/session.py", line 97, in wrapper
    return func(*args, session=session, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 3005, in _run_raw_task
    return _run_raw_task(
           ^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 273, in _run_raw_task
    TaskInstance._execute_task_with_callbacks(
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 3159, in _execute_task_with_callbacks
    result = self._execute_task(context, task_orig)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 3183, in _execute_task
    return _execute_task(self, context, task_orig)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 767, in _execute_task
    result = _execute_callable(context=context, **execute_callable_kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 733, in _execute_callable
    return ExecutionCallableRunner(
           ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/utils/operator_helpers.py", line 252, in run
    return self.func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/baseoperator.py", line 417, in wrapper
    return func(self, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/providers/apache/spark/operators/spark_submit.py", line 176, in execute
    self._hook.submit(self.application)
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/providers/apache/spark/hooks/spark_submit.py", line 560, in submit
    raise AirflowException(
airflow.exceptions.AirflowException: Cannot execute: spark-submit --master spark://sparkingflow-spark-master-1:7077 --name arrow-spark --deploy-mode client jobs/python/wordcountjob.py. Error code is: 1.
[2024-12-08T21:40:30.975+0000] {local_task_job_runner.py:266} INFO - Task exited with return code 1
[2024-12-08T21:40:31.001+0000] {taskinstance.py:3895} INFO - 0 downstream tasks scheduled from follow-on schedule check
[2024-12-08T21:40:31.004+0000] {local_task_job_runner.py:245} INFO - ::endgroup::
